packages/ember-glimmer/tests/integration/helpers/custom-helper-test.js

```diff
@@ -22,4 +26,431 @@ moduleFor('Helpers test: custom helpers', class extends RenderingTest {
+
+    this.assertText('1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('1');
+
+    this.runTask(() => helper.recompute());
+
+    this.assertText('2');
+
+    assert.strictEqual(destroyCount, 0, 'destroy is not called on recomputation');
+  }
+
+  ['@htmlbars helper with arg can recompute a new value']() {
+    let destroyCount = 0;
+    let count = 0;
+    let helper;
+
+    this.registerHelper('hello-world', {
+      init() {
+        this._super(...arguments);
+        helper = this;
+      },
+      compute() {
+        return ++count;
+      },
+      destroy() {
+        destroyCount++;
+        this._super();
+      }
+    });
+
+    this.render('{{hello-world "whut"}}');
+
+    this.assertText('1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('1');
+
+    this.runTask(() => helper.recompute());
+
+    this.assertText('2');
+
+    assert.strictEqual(destroyCount, 0, 'destroy is not called on recomputation');
+  }
+
+  ['@htmlbars shorthand helper is called for param changes']() {
+    let computeCount = 0;
+
+    this.registerHelper('hello-world', ([value]) => {
+      computeCount++;
+      return `${value}-value`;
+    });
+
+    this.render('{{hello-world name}}', {
+      name: 'bob'
+    });
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => set(this.context, 'name', 'sal'));
+
+    this.assertText('sal-value');
+
+    assert.strictEqual(computeCount, 2, 'compute is called exactly 2 times');
+
+    this.runTask(() => set(this.context, 'name', 'bob'));
+
+    this.assertText('bob-value');
+  }
+
+  ['@htmlbars helper compute is called for param changes']() {
+    let createCount = 0;
+    let computeCount = 0;
+
+    this.registerHelper('hello-world', {
+      init() {
+        this._super(...arguments);
+        // FIXME: Ideally, the helper instance does not need to be recreated
+        // for change of params.
+        createCount++;
+      },
+      compute([value]) {
+        computeCount++;
+        return `${value}-value`;
+      }
+    });
+
+    this.render('{{hello-world name}}', {
+      name: 'bob'
+    });
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => set(this.context, 'name', 'sal'));
+
+    this.assertText('sal-value');
+
+    assert.strictEqual(computeCount, 2, 'compute is called exactly 2 times');
+
+    this.runTask(() => set(this.context, 'name', 'bob'));
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(createCount, 1, 'helper is only created once');
```

Comment:
Probably should also assert the computeCount in addition to the createCount here